### PR TITLE
test: update animation click test

### PR DESCRIPTION
**Pull request by dgozman**

### CODE REVIEW

**Comment by JoelEinbinder** [comment]
What is this change doing? The test is still disabled?

**Comment by dgozman** [comment]
> What is this change doing? The test is still disabled?

Yep. The test was flaky by itself, so I updated it in the hopes we can now pass it, but that revealed two issues (see the comment).

**Comment by JoelEinbinder** [comment]
Is there a plan to ever pass this test?

File: test/click.spec.js [hunk 1]
```diff
@@ -404,34 +404,48 @@ module.exports.describe = function({testRunner, expect, playwright, FFOX, CHROMI
...
+        <div style="border: 1px solid black; height: 50px; overflow: auto; width: ${containerWidth}px;">
+        <button style="border: none; height: ${buttonSize}px; width: ${buttonSize}px; transition: left ${transition}ms linear 0s; left: 0; position: relative" onClick="window.clicked++"></button>
+        </div>
+        </body>
+        <script>
+          window.atLeft = true;
+          const animateLeft = () => {
+            const button = document.querySelector('button');
+            button.style.left = window.atLeft ? '${containerWidth - buttonSize}px' : '0px';
+            console.log('set ' + button.style.left);
```
Comment (JoelEinbinder) [inline test/click.spec.js]: nit: unneeded console log

File: test/click.spec.js [hunk 2]
```diff
@@ -404,34 +404,48 @@ module.exports.describe = function({testRunner, expect, playwright, FFOX, CHROMI
       expect(await page.evaluate('window.clicked')).toBe(true);
     });
     xit('should click on an animated button', async({page}) => {
```
Comment (JoelEinbinder) [inline test/click.spec.js]: nit: button animated by javascript.
CSS animations would be a very different thing.

File: test/click.spec.js [hunk 3]
```diff
@@ -404,34 +404,48 @@ module.exports.describe = function({testRunner, expect, playwright, FFOX, CHROMI
...
       `);
-      await page.click('button');
-      expect(await page.evaluate('window.clicked')).toBe(1);
-      expect(await page.evaluate('document.querySelector("#button").style.left')).toBe(`${containerWidth - buttonSize}px`);
-      await new Promise(resolve => setTimeout(resolve, 500));
-      await page.click('button');
-      expect(await page.evaluate('window.clicked')).toBe(2);
-      expect(await page.evaluate('document.querySelector("#button").style.left')).toBe('0px');
+      await page.evaluate(transition => {
+        window.setInterval(animateLeft, transition);
```
Comment (JoelEinbinder) [inline test/click.spec.js]: If we care specifically about the rafs, then this should probably be animated on raf right?
I'm not sure what we are testing, but if I was reviewing this code for a website I would say this is broken. You should always animate on raf, not setInterval.

File: test/click.spec.js (hunk 2 above)
Comment (JoelEinbinder) [reply test/click.spec.js]: Turns out this is CSS animations AND setInterval

Comment (JoelEinbinder) [inline test/click.spec.js]:
```suggestion
    xit('should fail to click a button animated via CSS animations and setInterval', async({page}) => {
```

File: test/click.spec.js [hunk 4]
```diff
@@ -404,34 +404,48 @@ module.exports.describe = function({testRunner, expect, playwright, FFOX, CHROMI
       expect(await page.evaluate('window.clicked')).toBe(true);
     });
     xit('should click on an animated button', async({page}) => {
-      const buttonSize = 50;
+      // This test exposes two issues:
```
Comment (JoelEinbinder) [inline test/click.spec.js]:
```suggestion
      // This test has a setInterval that consistently animates a button.
      // It checks that we detect the button to be continuously animating, and never try to click it.
      // This test exposes two issues:
```

**Pull request merged**